com.unity.netcode.gameobjects/Runtime/Serialization/BitWriter.cs

```diff
@@ -0,0 +1,210 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// Helper class for doing bitwise writes for a FastBufferWriter.
+    /// Ensures all bitwise writes end on proper byte alignment so FastBufferWriter doesn't have to be concerned
+    /// with misaligned writes.
+    /// </summary>
+    public ref struct BitWriter
+    {
+        private Ref<FastBufferWriter> m_Writer;
+        private unsafe byte* m_BufferPointer;
+        private readonly int m_Position;
+        private int m_BitPosition;
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+        private int m_AllowedBitwiseWriteMark;
+#endif
+        private const int k_BitsPerByte = 8;
+
+        /// <summary>
+        /// Whether or not the current BitPosition is evenly divisible by 8. I.e. whether or not the BitPosition is at a byte boundary.
+        /// </summary>
+        public bool BitAligned
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get => (m_BitPosition & 7) == 0;
+        }
+
+        internal unsafe BitWriter(ref FastBufferWriter writer)
+        {
+            m_Writer = new Ref<FastBufferWriter>(ref writer);
+            m_BufferPointer = writer.BufferPointer + writer.PositionInternal;
+            m_Position = writer.PositionInternal;
+            m_BitPosition = 0;
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+            m_AllowedBitwiseWriteMark = (m_Writer.Value.AllowedWriteMark - m_Writer.Value.Position) * k_BitsPerByte;
+#endif
+        }
+
+        /// <summary>
+        /// Pads the written bit count to byte alignment and commits the write back to the writer
+        /// </summary>
+        public void Dispose()
```

Comment:
well, `Dispose()` is often used to implement `IDisposable` interface in C# and it communicates something different than "pad bits", it's "free up resources" (which is often called from destructors when needed).
Should we rename this as `PadBits()` which would be similar to it's former predecessor `BitWriter` class?

Reply:
It can't be renamed. The point is that it's called automatically via `using()`. As I explained in a previous comment, ref structs cannot implement interfaces, but c# has special handling to support `using` syntax for any red structs that implement a `Dispose()` method.

Reply:
huh, interesting. OK :)